### ISSUE TITLE
New rule proposal: UseTargetTypedNewExpression

### ISSUE THREAD

**Issue by verdie-g** [DotNetAnalyzers/StyleCopAnalyzers#3388]
Use C# 9 [target-typed new expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new).

Bad
```csharp
List<int> l = new List<int>();
List<int> l = new List<int> { 1 };
var l = new List<int>();
```

Good
```csharp
List<int> l = new();
List<int> l = new() { 1 };
```

**Comment by sharwell** [comment]
This seems like an unnecessary duplicate of [IDE0090 (Simplify `new` expression)](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0090), which ships in the .NET SDK and can already be used both in the IDE and from the command line.

**Comment by verdie-g** [comment]
Indeed. For some reason this doesn't show up in Rider.
